processor/src/stack/mod.rs

```diff
@@ -161,13 +161,38 @@ impl Stack {
     /// stack will be at the last position in the word.
     ///
     /// Creating a word does not change the state of the stack.
-    pub fn get_word(&self, word_idx: usize) -> Word {
-        let offset = word_idx * WORD_SIZE;
+    pub fn get_word(&self, idx: usize) -> Word {
+        [self.get(idx + 3), self.get(idx + 2), self.get(idx + 1), self.get(idx)].into()
+    }
+
+    /// Returns the value at the specified position on the stack, including overflow items.
+    ///
+    /// This method can access items beyond the top 16 positions by reading from the overflow table.
+    /// Position 0 is the top of the stack, positions 0-15 are in the trace, and positions 16+ are
+    /// in the overflow table.
+    pub fn get_with_overflow(&self, pos: usize) -> Felt {
+        if pos < MIN_STACK_DEPTH {
+            // Item is in the trace (top 16 positions)
+            self.trace.get_stack_value_at(self.clk, pos)
+        } else {
+            // Item is in the overflow table
+            // Calculate the index within the overflow table
+            let overflow_index = pos - MIN_STACK_DEPTH;
+            self.overflow.get_element_at(overflow_index).unwrap_or(ZERO)
+        }
+    }
+
+    /// Returns a word from the stack at the specified word index, including overflow items.
+    ///
+    /// This method can access words that span into the overflow table.
+    /// Word 0 is defined by stack positions 0-3, word 1 by positions 4-7, etc.
+    /// The words are created in reverse order (top element at the last position).
+    pub fn get_word_with_overflow(&self, start_idx: usize) -> Word {
```

Comment:
Similar comment as above - maybe there could be just a single function for getting words from the stack and it would handle both cases.